### DISCUSSION TITLE
Context data reseting

### DISCUSSION THREAD

**Discussion by DZsolt01** [vercel/next.js#44537] [Help]
Hello!

I'm trying to make a global data with Context, but somehow it's reseting.

OrderContext.js

```
import { createContext, useState } from 'react'

const OrderContext = createContext();

export function OrderProvider({ children }) {
    const [order, setOrder] = useState("test");

    return (
        <OrderContext.Provider
            value={{
                order,
                setOrder
            }}
        >
            {children}
        </OrderContext.Provider>
    )
}
export default OrderContext;
```

_app.js
...
```

function MyApp({
  Component,
  pageProps: { session, ...pageProps },
}) {
  return (
    <SessionProvider session={session}>
      <ChakraProvider theme={theme}>
        <Layout>
          <OrderProvider>
            <Component {...pageProps} />
          </OrderProvider>
        </Layout>
      </ChakraProvider>
    </SessionProvider>
  )

}
```

index.js - it's the same in other pages too, i'm getting and seting the data succesfully, but it's reseting to default value.

`const { order, setOrder } = useContext(OrderContext)`

When i'm navigate between pages my data is resetting somehow.
I used Link for navigate.
navigation
`<Link href="/menu"><Button variant="ghost" color="white">Menu</Button></Link>`

**Comment by icyJoseph** [comment]
How do you calculate the layout? I suspect that either SessionProvider, Chakra or Layout, are remounting their children on route change. Hard to tell how right now, but it ought to be something like that.

Suppose a route change triggers some checks that in turn make either of those Providers or Layout, to return `null` for a frame, and then render their children again. Then the context would reset.

Try moving the OrderProvider, higher up on the tree (if it makes sense according to your business logic) and see if it persists.
